.github/workflows/integrate.yaml

```diff
@@ -50,11 +53,23 @@ jobs:
         sg microk8s -c 'juju bootstrap microk8s uk8s'
         juju add-model kubeflow
 
-    - name: Deploy istio-pilot
+    - name: Deploy charm dependencies
       run: |
         set -eux
-        juju deploy cs:istio-pilot
-        juju wait -wvt 300
+        juju model-config update-status-hook-interval=15s
+        juju deploy cs:istio
+        juju config istio-pilot default-gateway=kubeflow-gateway
+        sleep 10
+        kubectl patch role -n kubeflow istio-ingressgateway-operator -p '{"apiVersion":"rbac.authorization.k8s.io/v1","kind":"Role","metadata":{"name":"istio-ingressgateway-operator"},"rules":[{"apiGroups":["*"],"resources":["*"],"verbs":["*"]}]}'
+        juju deploy cs:kubeflow-dashboard --config profile=kubeflow-user
+        juju deploy cs:admission-webhook
+        juju deploy cs:kubeflow-profiles
+        juju relate kubeflow-dashboard kubeflow-profiles
+
+        # https://bugs.launchpad.net/juju/+bug/1921739
+        (i=60; while ! juju wait -wvt 300 ; do ((--i)) || exit; sleep 1; done)
+
+        juju model-config update-status-hook-interval=5m
```

Comment:
This should probably be converted to pytest-operator.

Reply:
Mind if I kick that off to another PR? This is already a little bigger than I was planning

Reply:
Sure!